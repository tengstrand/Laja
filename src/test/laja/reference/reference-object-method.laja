#write "target/test-output/reference-object-method.out"
  #set (x = new net.sf.laja.integrationtest.ReferenceTest$Data())
  #set (list = [1,2,3])
  #set (abc = "abc")
The value of x.valid = {x.valid}
The value of x.stringValue is: {x.stringValue}
abc.charAt(2) = {abc.charAt(2)}
x.stringValue.length = {x.stringValue.length}
x.array(1)[2] = {x.array(1)[2]}
x.list[1] = {x.list[1]}
x.method(123, "b", { a:"a1" b:"b1"}) = {x.method(123, "b", { a:"a1" b:"b1"} )}
#end
