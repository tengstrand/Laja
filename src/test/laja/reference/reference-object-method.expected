The value of x.valid = true
The value of x.stringValue is: a string value
abc.charAt(2) = c
x.stringValue.length = 14
x.array(1)[2] = 6
x.list[1] = two
x.method(123, "b", { a:"a1" b:"b1"}) = b1:123
